### PR TITLE
[Rewards] Moves configs to NPMRC, consolidates util files.

### PR DESCRIPTION
**Pull request by szilardszaloki**
Resolves https://github.com/brave/brave-browser/issues/20109.
DevOps issues: https://github.com/brave/devops/issues/9664, https://github.com/brave/devops/issues/9665.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
For __all three__ supported wallet providers, on __all three__ environments — make sure:
- you can link with the wallet provider
- the balance displays correctly
- you can (one-time) tip someone and the corresponding fee transaction goes through
- the account and activity URLs work fine

### CODE REVIEW

File: components/brave_rewards/core/gemini/gemini_util.h [hunk 1]
```diff
@@ -7,30 +7,34 @@
...
 std::string GetFeeAddress();
 
-std::string GetLoginUrl(const std::string& state);
+mojom::ExternalWalletPtr GenerateLinks(mojom::ExternalWalletPtr wallet);
 
-std::string GetAccountUrl();
+}  // namespace brave_rewards::internal::gemini
 
-std::string GetActivityUrl();
+namespace brave_rewards::internal::endpoint::gemini {
```
[the remark this answers is not in the document]
Comment (szilardszaloki) [reply components/brave_rewards/core/gemini/gemini_util.h:29]: Actually, this is intentionally not introducing a different namespace - this is where these stuff reside at the moment (only the file changed).

[the remark this answers is not in the document]
Comment (szilardszaloki) [reply components/brave_rewards/core/gemini/gemini_util.h:29]: Oh, and yes. I'll abstract away all these util classes in a follow-up. That's an intermediate step.

**Review by emerick — APPROVED** [review]

**Review by bsclifton — APPROVED** [review]
Looks great!

**Review by mihaiplesa — APPROVED** [review]

**Review by bridiver — APPROVED** [review]

File: components/brave_rewards/browser/test/rewards_state_browsertest.cc [hunk 2]
```diff
@@ -410,11 +410,12 @@ class V10 : public RewardsStateBrowserTest,
 };
 
 #ifdef OFFICIAL_BUILD
-#define _UPHOLD_CLIENT_ID_ +std::string(BUILDFLAG(UPHOLD_CLIENT_ID)) +
-#define _UPHOLD_URL_ "https://uphold.com"
+#define _UPHOLD_CLIENT_ID_ \
```
Comment (bridiver) [inline components/brave_rewards/browser/test/rewards_state_browsertest.cc:413]: why is there a leading and trailing underscore for these? That doesn't follow any chromium standard afaik

**Pull request merged**